core/src/main/scala/org/locationtech/rasterframes/expressions/BinaryRasterFunction.scala

```diff
@@ -25,13 +25,14 @@ import com.typesafe.scalalogging.Logger
 import geotrellis.raster.Tile
 import org.apache.spark.sql.catalyst.analysis.TypeCheckResult
 import org.apache.spark.sql.catalyst.analysis.TypeCheckResult.{TypeCheckFailure, TypeCheckSuccess}
-import org.apache.spark.sql.catalyst.expressions.BinaryExpression
+import org.apache.spark.sql.catalyst.expressions.{BinaryExpression, Expression}
 import org.apache.spark.sql.types.DataType
 import org.locationtech.rasterframes.expressions.DynamicExtractors._
 import org.slf4j.LoggerFactory
 
 /** Operation combining two tiles or a tile and a scalar into a new tile. */
-trait BinaryRasterFunction extends BinaryExpression with RasterResult {
+trait BinaryRasterFunction extends BinaryExpression with RasterResult { self: HasBinaryExpressionCopy =>
+  override protected def withNewChildrenInternal(newLeft: Expression, newRight: Expression): Expression = copy(newLeft, newRight)
```

Comment:
Tried being clever here with structural self type in order to call copy in the super-type and I really don't think this is worth the brain cells. Down this road is creating things like `RFBinaryExpression` that has this little bit of magic. However, it will not always work, for instance when the case class does not use standard `left`, `right` or `child` constructor names. So I'm probably going to revert this and do some more c/p.

Reply:
Yes, this was a bad idea